src/devices/machine/e05a30.cpp

```diff
@@ -167,6 +168,15 @@ void e05a30_device::write(offs_t offset, uint8_t data)
 	LOG("%s: e05a30_w([0xC0%02x]): %02x\n", machine().describe_context(), offset, data);
 
 	switch (offset) {
+	case 0x00:
+		m_c000_shift_register = 0;
+		m_c000_read = 0;
```

Comment:
This is a change in behaviour – you’re now clearing the shift register on a write to offset zero.  Is that correct?  Also, you’re apparently missing a `break` statement here.